source/crud/tests/README.rst

```diff
@@ -12,29 +12,17 @@ Introduction
 The YAML and JSON files in this directory tree are platform-independent tests
 that drivers can use to prove their conformance to the CRUD spec.
 
-Given the variety of languages and implementations and limited nature of a
-description of a test, there are a number of things that aren't testable. For
-instance, none of these tests assert that maxTimeMS was properly sent to the
-server. This would involve a lot of infrastructure to define and setup.
-Therefore, these YAML tests are in no way a replacement for more thorough
-testing. However, they can provide an initial verification of your
-implementation.
-
 Running these integration tests will require a running MongoDB server or
 cluster with server versions 2.6.0 or later. Some tests have specific server
 version requirements as noted by ``minServerVersion`` and ``maxServerVersion``.
 
-Version
-=======
-
-Files in the "specifications" repository have no version scheme. They are not
-tied to a MongoDB server version.
-
-Format
-======
+Test Format
+===========
 
 Each YAML file has the following keys:
 
+- ``collection_name`` (optional): The collection to use for testing.
```

Comment:
I will need to add `collectionOptions` to this in order to write tests for https://jira.mongodb.org/browse/SPEC-1153. Would you be in favor of adding optional fields for `collectionOptions` and `databaseOptions`  (both would be arrays of documents)? This would be added under `tests` of course.

Reply:
The transaction tests have collectionOptions placed on the same level as 'object' like this:
```
  - description: collection writeConcern without transaction
    operations:
      - name: insertOne
        object: collection
        collectionOptions:
          writeConcern:
            w: majority
        arguments:
```
https://github.com/mongodb/specifications/tree/master/source/transactions/tests#test-format

Reply:
The Transaction spec defines a `clientOptions` key under the `tests` field, which I omitted here because the CRUD specs had no use for it. I think it'd make sense to add it for completeness' sake, though.

As for `collectionOptions`, Transactions already defines that as a field within the operation document (e.g. `tests.operations[i]`). Would that work for you?

The only other relevant spec I can think of with its own deviating test format is Change Streams, and that does not define `clientOptions` or `collectionOptions` anywhere (yet), so we don't have to worry about potentially introducing a new conflict.

Reply:
I missed `clientOptions`, but I think we should definitely add it here as well.

Having a `collectionOptions` field at the same level as `object` would work perfectly. It also makes sense to put it there so that it can be changed for different test cases.

Reply:
Added both to the format. Feel free to resolve.